File: src/components/pages/projects/$projectId/shared/connect-snippets/deno/fresh/routes/v1/policies.ts
import { Handlers } from "$fresh/server.ts"
import { Project } from "{{DENO_SDK_SPECIFIER}}"
import { client } from "../../lib/appwrite.ts"

const project = new Project(client)

export const handler: Handlers = {
  async PATCH() {
    const policy = await project.updatePasswordStrengthPolicy({
      min: 8,
      uppercase: true,
      number: true,
      symbols: true
    })
    return Response.json(policy)
  },
  async GET() {
    return Response.json(await project.listPolicies())
  },
}

File: src/components/pages/projects/$projectId/shared/connect-snippets/dart/frog/routes/v1/policies.dart
import 'package:dart_appwrite/dart_appwrite.dart' hide Response;
import 'package:dart_frog/dart_frog.dart';

Future<Response> onRequest(RequestContext context) async {
  final project = context.read<Project>();

  if (context.request.method == HttpMethod.patch) {
    final policy = await project.updatePasswordStrengthPolicy(
      min: 8,
      uppercase: true,
      number: true,
      symbols: true,
    );

    return Response.json(body: policy.toMap());
  }

  final policies = await project.listPolicies();

  return Response.json(body: policies.toMap());
}
